src/core/utils/components/MiddleTextTruncation.tsx

```diff
@@ -0,0 +1,61 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { useOverflow } from '../hooks/useOverflow';
+import { CommonProps } from '../props';
+
+const ELLIPSIS_CHAR = '…';
+
+export type MiddleTextTruncationProps = {
+  /**
+   * Text to truncate.
+   */
+  text: string;
+  /**
+   * Number of characters to leave at the end.
+   * @default 6
+   */
+  endCharsCount?: number;
+} & CommonProps;
+
+/**
+ * Truncates text with the ellipsis in the middle,
+ * leaving defined number of chars at the end.
+ * @example
+ * <MiddleTextTruncation text='Here can be a very long text' />
+ */
+export const MiddleTextTruncation = (props: MiddleTextTruncationProps) => {
+  const { text, endCharsCount = 6, style, ...rest } = props;
+
+  const [ref, visibleCount] = useOverflow(text);
+
+  const truncatedText = React.useMemo(() => {
+    if (visibleCount < text.length) {
+      return `${text.substr(
+        0,
+        visibleCount - endCharsCount - ELLIPSIS_CHAR.length,
```

Comment:
`ELLIPSIS_CHAR.length` will always be 1 so what's the use

Reply:
whats the use of what?

Reply:
what's the use of `ELLIPSIS_CHAR.length`? It will always be 1 because it is just one character 😆